File: 05-singleton/step1/java/Main.java
package step1;

public class Main {
    public static void main(String[] args) {
        SingletonExample.GetInstance().Do();

        Caller caller1 = new Caller();
        Caller caller2 = new Caller();

        SingletonExample.GetInstance().value = 10;

        caller1.Do();
        caller2.Do();

        SingletonExample.GetInstance().value = 200;

        caller1.Do();
        caller2.Do();
    }
}

File: 05-singleton/step1/kotlin/Main.kt
package step1;

fun main(){
    SingletonExample.getInstance().Do()

    val caller1 = Caller()
    val caller2 = Caller()

    SingletonExample.getInstance().value = 10f

    caller1.Do()
    caller2.Do()

    SingletonExample.getInstance().value = 200f

    caller1.Do()
    caller2.Do()
}